gnovm/pkg/gnolang/preprocess.go

```diff
@@ -4276,6 +4276,12 @@ func tryPredefine(store Store, last BlockNode, d Decl) (un Name) {
 				if isBlankIdentifier(tx) {
 					panic("cannot use _ as value or type")
 				}
+
+				// do not allow nil as type.
+				if tx.Name == "nil" {
+					panic("nil is not a type")
+				}
```

Comment:
WDYT, handle other bultin names like len,cap.. ?
```suggestion
				// do not allow type aliasing of builtin names
				if isUverseName(tx.Name) {
					panic(fmt.Sprintf("%s is not a type", tx.Name))
				}
```